src/App.js

```diff
@@ -14,11 +15,137 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    isVisible: true,
+    lengthLimit: 1,
+    goods: goodsFromServer,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  isVisible = () => {
+    this.setState({
+      isVisible: false,
+    });
+  }
+
+  changeLength = (event) => {
+    this.setState(
+      { lengthLimit: +event.target.value },
+    );
+  }
+
+  reset = () => {
+    this.setState({
+      lengthLimit: 1,
+      sortBy: '',
+      isReversed: false,
+    });
+  }
+
+  reverseGoods = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  sortBylength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  sortAlphabetically = () => {
+    this.setState({ sortBy: 'alphabetical' });
+  }
+
+  render() {
+    const { isVisible, lengthLimit, goods, sortBy, isReversed } = this.state;
+    const select = new Array(goodsFromServer.length).fill(1);
+
+    const visibleGoods = goods.filter(good => good.length >= lengthLimit);
+
+    visibleGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case 'length':
+          return good1[sortBy] - good2[sortBy];
+
+        case 'alphabetical':
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+
+        {isVisible && (
```

Comment:
```suggestion
        {!isVisible && (
```
isVisible думаю скорее относится к списку goods, а не к кнопке start